janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchScroll.java

```diff
@@ -28,19 +28,28 @@
  */
 public class ElasticSearchScroll implements Iterator<RawQuery.Result<String>> {
 
-    private final BlockingQueue<RawQuery.Result<String>> queue;
+    private final Queue<Result<String>> queue;
     private boolean isFinished;
     private final ElasticSearchClient client;
     private final String scrollId;
     private final int batchSize;
 
     public ElasticSearchScroll(ElasticSearchClient client, ElasticSearchResponse initialResponse, int nbDocByQuery) {
-        queue = new LinkedBlockingQueue<>();
+        queue = new ArrayDeque<>();
```

Comment:
Why did you change this to non blocking queue? What about thread safety?

Reply:
I could undo this change as I have fixes the problem in a better way by calling hasNext in the next method to make sure that if a loop is just calling next that it reads all records.
I don't believe that this is used from multiple threads though.

Reply:
Reverted this part of the commit